src/modules.ts

```diff
@@ -209,108 +217,87 @@ function isTheServerModule(moduleName: string, config: Config) {
-      Date.now() - modulesByKeyword[keyword].time < 60 * 1000
-    ) {
-      resolve(modulesByKeyword[keyword].packages)
-      return
-    }
+  const moduleData = await searchByKeyword(keyword)
+  npmDebug(
+    `npm search returned ${moduleData.length} modules with keyword ${keyword}`
+  )
 
-    searchByKeyword(keyword)
-      .then((moduleData) => {
-        npmDebug(
-          `npm search returned ${moduleData.length} modules with keyword ${keyword}`
-        )
-        const result = moduleData.reduce(
-          (
-            acc: { [packageName: string]: NpmModuleData },
-            module: NpmModuleData
-          ) => {
-            const name = module.package.name
-            if (
-              !acc[name] ||
-              semver.gt(module.package.version, acc[name].package.version)
-            ) {
-              acc[name] = module
-            }
-            return acc
-          },
-          {}
-        )
-        const packages = _.values(result)
-        modulesByKeyword[keyword] = {
-          time: Date.now(),
-          packages
-        }
-        resolve(packages)
-      })
-      .catch((e) => {
-        reject(e)
-      })
-  })
+  const result = moduleData.reduce(
+    (acc: Record<string, NpmModuleData>, module: NpmModuleData) => {
+      const name = module.package.name
+      if (
+        !acc[name] ||
+        semver.gt(module.package.version, acc[name].package.version)
+      ) {
+        acc[name] = module
+      }
+      return acc
+    },
+    {}
+  )
+
+  const packages = Object.values(result)
+  modulesByKeyword[keyword] = { time: Date.now(), packages }
+  return packages
 }
 
-function searchByKeyword(keyword: string): Promise<NpmModuleData[]> {
-  return new Promise((resolve, reject) => {
-    let fetchedCount = 0
-    let toFetchCount = 1
-    let moduleData: NpmModuleData[] = []
-    const npmFetch = () => {
-      npmDebug(
-        `searching ${keyword} from ${fetchedCount + 1} of ${toFetchCount}`
-      )
-      fetch(
-        `https://registry.npmjs.org/-/v1/search?size=250&from=${
-          fetchedCount > 0 ? fetchedCount : 0
-        }&text=keywords:${keyword}`
-      )
-        .then((r) => r.json())
-        .then((parsed) => {
-          moduleData = moduleData.concat(parsed.objects)
-          fetchedCount += parsed.objects.length
-          toFetchCount = parsed.total
-          if (fetchedCount < toFetchCount) {
-            nextTick(() => npmFetch())
-          } else {
-            resolve(moduleData)
-          }
-        })
-        .catch(reject)
-    }
-    npmFetch()
-  })
+async function searchByKeyword(keyword: string): Promise<NpmModuleData[]> {
+  let fetchedCount = 0
+  let toFetchCount = 1
+  let moduleData: NpmModuleData[] = []
+
+  while (fetchedCount < toFetchCount) {
+    npmDebug(`searching ${keyword} from ${fetchedCount + 1} of ${toFetchCount}`)
+    const res = await fetch(
+      `https://registry.npmjs.org/-/v1/search?size=10&from=${
```

Comment:
Why did you change size to 10? Appstore no longer works, because with the small windowsize npm throws Too many requests after rapidly firing some requests - with content that is not json that we fail to parse.

Reply:
Reverted to 250 in 07bc9a9fa08f93ad160e209beb2033ad464fa560